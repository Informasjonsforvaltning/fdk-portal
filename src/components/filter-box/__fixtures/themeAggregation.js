export const themeAggregation = {
  buckets: [
    { key: 'GOVE', count: 104 },
    { key: 'EDUC', count: 11 },
    { key: 'HEAL', count: 8 },
    { key: 'Ukjent', count: 7 },
    { key: 'AGRI', count: 1 },
    { key: 'TEST', count: 1 }
  ]
};
