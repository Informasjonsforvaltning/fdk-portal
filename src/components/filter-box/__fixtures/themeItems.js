export const themeItems = {
  AGRI: {
    id: 'http://publications.europa.eu/resource/authority/data-theme/AGRI',
    code: 'AGRI',
    startUse: '2015-10-01',
    title: {
      pt: 'Agricultura, pesca, silvicultura e alimentação',
      bg: 'Селско стопанство, рибарство, горско стопанство и храни'
    },
    conceptSchema: {
      id: 'http://publications.europa.eu/resource/authority/data-theme',
      title: { en: 'Dataset types Named Authority List' }
    }
  }
};
