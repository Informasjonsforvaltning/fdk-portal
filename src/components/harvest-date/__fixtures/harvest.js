export const harvest = {
  firstHarvested: '2018-06-25T19":51:05+0200',
  lastHarvested: '2018-07-09T14:40:37+0200',
  lastChanged: '2018-07-09T14:40:37+0200',
  changed: [
    '2018-06-25T19:51:05+0200',
    '2018-06-25T19:51:12+0200',
    '2018-06-26T09:16:46+0200',
    '2018-06-28T09:22:41+0200',
    '2018-06-28T09:52:24+0200',
    '2018-06-29T15:25:53+0200',
    '2018-07-03T10:59:38+0200',
    '2018-07-09T14:40:37+0200'
  ]
};
