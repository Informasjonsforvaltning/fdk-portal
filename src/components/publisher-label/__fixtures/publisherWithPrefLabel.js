export const publisherWithPrefLabel = {
  type: 'no.dcat.datastore.domain.dcat.Publisher',
  valid: false,
  uri: 'http://data.brreg.no/enhetsregisteret/enhet/910244132',
  id: '910244132',
  name: 'RAMSUND OG ROGNAN REVISJON',
  orgPath: '/ANNET/910244132',
  prefLabel: {
    no: 'Ramsund og Rognan revisjon'
  }
};
