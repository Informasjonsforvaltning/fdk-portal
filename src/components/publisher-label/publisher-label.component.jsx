import React from 'react';
import PropTypes from 'prop-types';
import _ from 'lodash';
import { getTranslateText } from '../../lib/translateText';
import localization from '../../lib/localization';

const displayCatalogTitle = (publisherItem, catalog) => {
  if (
    catalog &&
    _.get(publisherItem, 'id') !== _.get(catalog, ['publisher', 'id'])
  ) {
    const catalogTitle = getTranslateText(_.get(catalog, ['title']));

    return ` (${localization.dataset.registeredIn} ${catalogTitle})`;
  }

  return null;
};

export const PublisherLabel = props => {
  const { label, tag: Tag, publisherItem, catalog } = props;
  if (!publisherItem) {
    return null;
  }

  const publisherPrefLabel =
    getTranslateText(_.get(publisherItem, ['prefLabel'])) ||
    _.capitalize(_.get(publisherItem, 'name', ''));

  return (
    <span className="mr-3">
      {label}&nbsp;
      <Tag>
        {publisherPrefLabel}
        {displayCatalogTitle(publisherItem, catalog)}
      </Tag>
    </span>
  );
};

PublisherLabel.defaultProps = {
  label: null,
  tag: 'span',
  publisherItem: null
};

PublisherLabel.propTypes = {
  label: PropTypes.string,
  tag: PropTypes.string,
  publisherItem: PropTypes.object
};
