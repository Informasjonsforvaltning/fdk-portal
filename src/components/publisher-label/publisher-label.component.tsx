import React, { FC } from 'react';
import capitalize from 'lodash/capitalize';

import { getTranslateText } from '../../lib/translateText';
import localization from '../../lib/localization';

interface Props {
  label?: string;
  tag?: any;
  publisherItem?: Record<string, any>;
  catalog: any;
}

export const PublisherLabel: FC<Props> = ({
  label,
  tag: Tag = 'span',
  publisherItem,
  catalog
}) => {
  const publisherPrefLabel =
    getTranslateText(publisherItem?.prefLabel) ||
    capitalize(publisherItem?.name ?? '');

  const catalogTitle =
    publisherItem?.id !== catalog?.publisher?.id
      ? ` (${localization.dataset.registeredIn} ${getTranslateText(
          catalog?.title
        )})`
      : null;

  return publisherItem ? (
    <span className='mr-3'>
      {label}&nbsp;
      <Tag>
        {publisherPrefLabel}
        {catalogTitle}
      </Tag>
    </span>
  ) : null;
};
