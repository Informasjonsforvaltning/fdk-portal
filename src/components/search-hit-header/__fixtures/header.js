export const header = {
  title: 'Test tittel',
  publisherLabel: 'Eies av',
  publisher: {
    id: '974760673'
  }
};
