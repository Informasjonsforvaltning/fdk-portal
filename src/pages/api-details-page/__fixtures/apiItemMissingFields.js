export const apiItemMissingFields = {
  id: 'c541c305-4e4f-43ed-be3e-5434740ec1b5',
  uri: 'jar:file:/app.jar!/BOOT-INF/classes!/datakatalog-api.json',
  apiSpecification: {
    info: {
      description:
        'Provides a basic search api against the National Data Directory of Norway',
      version: '1.0',
      title: 'National Data Directory Search API',
      termsOfService: 'https://fellesdatakatalog.brreg.no/about',
      contact: {
        name: 'Brønnøysundregistrene',
        url: 'https://fellesdatakatalog.brreg.no',
        email: 'fellesdatakatalog@brreg.no'
      },
      license: {
        name: 'License of API',
        url: 'http://data.norge.no/nlod/no/2.0'
      }
    }
  }
};
