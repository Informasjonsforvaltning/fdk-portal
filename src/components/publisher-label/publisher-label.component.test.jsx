import React from 'react';
import { shallow } from 'enzyme';
import { PublisherLabel } from './publisher-label.component';
import { publisherWithPrefLabel } from './__fixtures/publisherWithPrefLabel';
import { publisherWithNoPrefLabel } from './__fixtures/publisherWithNoPrefLabel';

test('should render PublisherLabel with publisher', () => {
  const defaultProps = {
    label: 'Eies av',
    publisherItem: publisherWithPrefLabel
  };
  const wrapper = shallow(<PublisherLabel {...defaultProps} />);
  expect(wrapper).toMatchSnapshot();
});

test('should render PublisherLabel with no props', () => {
  const wrapper = shallow(<PublisherLabel />);
  expect(wrapper).toMatchSnapshot();
});

test('should render PublisherLabel with no prefLabel', () => {
  const defaultProps = {
    label: 'Eies av',
    publisherItem: publisherWithNoPrefLabel
  };
  const wrapper = shallow(<PublisherLabel {...defaultProps} />);
  expect(wrapper).toMatchSnapshot();
});
