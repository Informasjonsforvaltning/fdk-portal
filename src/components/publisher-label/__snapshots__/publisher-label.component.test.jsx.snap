// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render PublisherLabel with no prefLabel 1`] = `
<span
  className="mr-3"
>
  Eies av
   
  <span>
    Ramsund og rognan revisjon
  </span>
</span>
`;

exports[`should render PublisherLabel with no props 1`] = `""`;

exports[`should render PublisherLabel with publisher 1`] = `
<span
  className="mr-3"
>
  Eies av
   
  <span>
    Ramsund og Rognan revisjon
  </span>
</span>
`;
